Backend/app/routes/notification.py

```
@@ -0,0 +1,210 @@
+            logger.error(f"Notification {notif_id} saved locally but failed to push to Supabase.")
+            return JSONResponse(status_code=202, content={"error": "Notification saved locally, but failed to push to Supabase. Realtime delivery may be delayed."})
+        logger.info(f"Notification {notif_id} created for user {user['id']}.")
+        return notif_dict
+    except Exception as e:
+        logger.error(f"Failed to create notification: {e}")
+        await db.rollback()
+        return JSONResponse(status_code=500, content={"error": f"Failed to create notification: {str(e)}"})
+
+@router.delete("/", status_code=status.HTTP_204_NO_CONTENT)
+async def delete_notifications(
+    ids: List[str] = Body(...),
```

Ruff (0.11.9)

Do not perform function call `Body` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)